apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -87,19 +92,31 @@ const InnerFileBrowser = React.memo(
                       moveFilePrompt={moveFilePrompt}
                       renameFilePrompt={renameFilePrompt}
                       renameFolderPrompt={renameFolderPrompt}
+                      toggleFileVisibility={toggleFileVisibility}
                     />
                   </ul>
                 )}
               </li>
             );
           })}
         {Object.values(files)
-          .filter(f => f.folderId === parentId)
+          .filter(
+            f => f.folderId === parentId && (f.visible !== false || isStartMode)
+          )
           .sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
Non-blocking: It seems like
```
Object.values(files)
          .filter(
            f => f.folderId === parentId && (f.visible !== false || isStartMode)
          )
          .sort((a, b) => a.name.localeCompare(b.name))
```

Could be a good candidate for extracting into a helper or separate variable, similar to how you did `getActiveFileForProject`?

Reply:
This is a little simpler now that I've renamed/reworked `.visible` to `.hidden`. PTAL